mobile/library/common/http/client.cc

```diff
@@ -406,54 +412,77 @@ void Client::DirectStream::saveLatestStreamIntel() {
 }
 
 void Client::DirectStream::saveFinalStreamIntel() {
+  if (!parent_.getStream(stream_handle_, ALLOW_ONLY_FOR_OPEN_STREAMS)) {
+    return;
+  }
   OptRef<RequestDecoder> request_decoder = requestDecoder();
-  if (!request_decoder || !parent_.getStream(stream_handle_, ALLOW_ONLY_FOR_OPEN_STREAMS)) {
+  if (!request_decoder) {
     return;
   }
   StreamInfo::setFinalStreamIntel(request_decoder->streamInfo(), parent_.dispatcher_.timeSource(),
                                   envoy_final_stream_intel_);
 }
 
-envoy_error Client::DirectStreamCallbacks::streamError() {
+void Client::DirectStreamCallbacks::latchError() {
+  if (error_.has_value()) {
+  }
   envoy_error error{};
+  error_ = error;
```

Comment:
it took me a minute to understand what's going on here. Could we perhaps do:
`error_ = envoy_error();`

Reply:
done